### PR TITLE
[connectivity] Add macos example

### PR DESCRIPTION
**Pull request by franciscojma86**
## Description

Adds an example app to run integration tests on CI
## Related Issues

*Replace this paragraph with a list of issues related to this PR from the [issue database](https://github.com/flutter/flutter/issues). Indicate, which of these issues are resolved or fixed by this PR. Note that you'll have to prefix the issue numbers with flutter/flutter#.*

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

### CODE REVIEW

File: packages/connectivity/connectivity_macos/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+## 0.1.0
+
+* Adds an example app to trigger CI tests. Bumped the MINOR version to
+avoid compatibility issues once this packages is endoresed.
```
Comment (ditman) [inline packages/connectivity/connectivity_macos/CHANGELOG.md]: Typo "endoresed" -> "endorsed"

File: packages/connectivity/connectivity_macos/README.md [hunk 2]
```diff
@@ -6,25 +6,27 @@ The macos implementation of [`connectivity`].
...
+dependency:
 
-Once you have the `connectivity_macos` dependency in your pubspec, you should
-be able to use `package:connectivity` as normal.
+```yaml
+...
+dependencies:
+  ...
+  connectivity_macos: ^0.0.2
+  ...
```
Comment (ditman) [inline packages/connectivity/connectivity_macos/README.md]: This is not needed (but if you want to keep it, the version of connectivity_macos should be `^0.1.0`.

We've also added links to the core plugin documentation for people to find instructions there:

* https://github.com/FirebaseExtended/flutterfire/blob/master/packages/cloud_firestore/cloud_firestore_web/README.md

File: packages/connectivity/connectivity_macos/example/macos/Flutter/GeneratedPluginRegistrant.swift [hunk 3]
```diff
@@ -0,0 +1,12 @@
+//
+//  Generated file. Do not edit.
```
Comment (ditman) [inline packages/connectivity/connectivity_macos/example/macos/Flutter/GeneratedPluginRegistrant.swift]: Does this file need to be committed?

Comment (franciscojma86) [reply packages/connectivity/connectivity_macos/example/macos/Flutter/GeneratedPluginRegistrant.swift]: nope, need to update the gitignore to include swift :)

Comment (franciscojma86) [reply packages/connectivity/connectivity_macos/example/macos/Flutter/GeneratedPluginRegistrant.swift]: https://github.com/flutter/plugins/pull/2540 :)

**Pull request merged**